### PR TITLE
feat(13421): create layer features panel

### PR DESCRIPTION
**Pull request by vkozio**
HOT Projects feature panel implementation (13421)

## Summary by CodeRabbit

- **New Features**
  - Introduced the Layer Features Panel for enhanced interaction with map layers.
  - Added new card components for displaying various elements like action buttons, labels, progress, and tables within the Layer Features Panel.
  - Implemented a new sorting and formatting functionality for hot projects in the Layer Features Panel.

- **Enhancements**
  - Added new styling for the Layer Features Panel and its components to improve user interface aesthetics and usability.

- **Integrations**
  - Integrated the Layer Features Panel into the Map view, enabling it conditionally based on feature flags.

### CODE REVIEW

File: src/features/layer_features_panel/atoms/hotProjects_outlines.ts
Comment (Akiyamka) [inline src/features/layer_features_panel/atoms/hotProjects_outlines.ts:1]: It is not clear why cammelCase and snake_case are mixed in the file name.

File: src/features/layer_features_panel/atoms/hotProjects_outlines.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+const formatTime = (d: string | number | Date) => formatTimeFn(new Date(d));
+
+function toTitleCase(word: string) {
+  return word.charAt(0).toUpperCase() + word.substring(1).toLowerCase();
+}
+
+const RED = { color: 'var(--error-strong)', backgroundColor: '#F8DDE0' }; // 'var(--error-weak)'
+const GREEN = { color: 'var(--success-strong)', backgroundColor: '#EAF6E6' }; // 'var(--success-weak)'
+const ORANGE = { color: 'var(--warning-strong)', backgroundColor: '#FFE8D4' }; // 'var(--warning-weak)'
+const DARKGREY = {
```
Comment (Akiyamka) [inline src/features/layer_features_panel/atoms/hotProjects_outlines.ts]: What do these comments mean?
// 'var(--error-weak)'
// 'var(--success-weak)'
// 'var(--warning-weak)'

File: src/features/layer_features_panel/atoms/hotProjects_outlines.ts [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  return { value: toTitleCase(s) + ' priority', ...(colorMap[s] || {}) };
+}
+
+/*
+Features should be sorted by their status (PUBLISHED → archived),
+then by priority (high → medium → low) and then by mapped % (low → high).
+HOT projects with high priority and low mapped % should be higher
+*/
+function hotProjectsSort(a, b) {
+  const weight_status = {
```
Comment (Akiyamka) [inline src/features/layer_features_panel/atoms/hotProjects_outlines.ts]: Please, be consistent in names casing  here and in other places

File: src/features/layer_features_panel/atoms/hotProjects_outlines.ts [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+                ]
+              : []),
+            buildPriorityLabel(p.projectPriority),
+          ],
+        },
+        {
+          type: 'title',
+          // icon: 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAACXBIWXMAAAsTAAALEwEAmpwYAAAAw0lEQVR4nO2WWwrCMBBFz0ezhoouxYK4IyNdkdbuR5qFiAtQKUyhP0rz0IyQA/nrzTxymwwUCn9ABTjg6bmcaKM5BgSflo0NXgM34AHsPHRb0dyBdUwCnVRyCtCeRTvuEUQTWcXUvTGJva+4Aq4iPhCOnRnShBjP+Qo/FGJ/0roUR9lJ8Avp6Jcaspllu0mYwGpJVw0wyEct6Wll7+Gdr0zuBLIfgQoTqvgNs19EKq7i7I+RiudYxUCSfSRTMZQWCnyTF73qivMHBsP9AAAAAElFTkSuQmCC',
+          title: `${p.projectInfo.name}`,
+          // subtitle: 'Sub',
```
Comment (Akiyamka) [inline src/features/layer_features_panel/atoms/hotProjects_outlines.ts]: Uncomment or remove

File: src/features/layer_features_panel/atoms/hotProjects_outlines.ts [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        },
+        {
+          type: 'progress',
+          // caption: 'Caption',
```
Comment (Akiyamka) [inline src/features/layer_features_panel/atoms/hotProjects_outlines.ts]: same

File: src/features/layer_features_panel/atoms/layerFeaturesCollectionAtom.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
+import { configRepo } from '~core/config';
+import { createAtom } from '~utils/atoms';
+import { focusedGeometryAtom } from '~core/focused_geometry/model';
+import { apiClient } from '~core/apiClientInstance';
+import { createNumberAtom } from '~utils/atoms/createPrimitives';
+import { enabledLayersAtom } from '~core/logical_layers/atoms/enabledLayers';
+import { getPanelData } from './hotProjects_outlines';
+import type { FeatureCardCfg } from '../components/CardElements';
+
+export const FEATURESPANEL_LAYER_ID = 'hotProjects_outlines';
```
Comment (Akiyamka) [inline src/features/layer_features_panel/atoms/layerFeaturesCollectionAtom.ts]: In other features we move that in separate file constants.ts

File: src/features/layer_features_panel/components/CardElements/Table.tsx [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+  return (
+    <div className={s.table}>
+      {rows.map((row, i) => (
+        <div key={i}>
+          {row.map((r, ri) => (
+            <div key={ri}>{r}</div>
+          ))}
+        </div>
+      ))}
+    </div>
```
Comment (Akiyamka) [inline src/features/layer_features_panel/components/CardElements/Table.tsx:13]: I wonder, why it's not a <table>?

**Review by Akiyamka — COMMENTED** [review]
there are no questions in realization, only questions on stylistic moments

File: src/features/layer_features_panel/components/index.ts
Comment (dqunbp) [inline src/features/layer_features_panel/components/index.ts:1]: Do we need the empty file?

File: src/features/layer_features_panel/components/LayerFeaturesPanel/index.tsx [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+  const isOpen = panelState !== 'closed';
+  const isShort = panelState === 'short';
+  const isMobile = useMediaQuery(IS_MOBILE_QUERY);
+
+  const handleRefChange = useHeightResizer(
+    (isOpen) => !isOpen && setPanelState('closed'),
+    isOpen,
+    MIN_HEIGHT,
+    'lf_list',
```
Comment (dqunbp) [inline src/features/layer_features_panel/components/LayerFeaturesPanel/index.tsx:41]: Is is required to use snake case here?

File: src/features/layer_features_panel/components/CardElements/index.ts [hunk 8]
```diff
@@ -0,0 +1,28 @@
+import { ActionButtons } from './ActionButtons';
+import { Title } from './Title';
+import { Table } from './Table';
+import { Label } from './Label';
+import { Progress } from './Progress';
+import type { LngLatBoundsLike } from 'maplibre-gl';
+
+// TODO: add new card elements here
```
Comment (dqunbp) [inline src/features/layer_features_panel/components/CardElements/index.ts]: Looks like we need to remove TODO here

File: src/features/layer_features_panel/components/CardElements/Title.tsx [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+export interface TitleProps {
+  icon?: string;
+  title: string;
+  subtitle?: string;
+}
+
+export function Title({ icon, title, subtitle }: TitleProps) {
+  return (
+    <div className={s.title}>
+      {icon && <img src={icon} alt="" />}
```
Comment (dqunbp) [inline src/features/layer_features_panel/components/CardElements/Title.tsx]: Can we use title as `alt` value?

File: src/features/layer_features_panel/components/CardElements/Progress.tsx [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+
+export function Progress({ caption, items }: ProgressProps) {
+  return (
+    <div className={s.progress}>
+      {caption && <div className={s.caption}>{caption}</div>}
+      <div className={s.stack}>
+        {items
+          .map(({ color, title, value }, i) => {
+            const style = {};
+            if (color) style['backgroundColor'] = color;
```
Comment (dqunbp) [inline src/features/layer_features_panel/components/CardElements/Progress.tsx:22]: Is it better to use css variables here?

File: src/features/layer_features_panel/components/CardElements/Progress.tsx [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+            style['width'] = value + '%';
+            return <div key={i} style={style} />;
+          })
+          .reverse()}
+      </div>
+      <div className={s.desc}>
+        {items.map(({ color, title, value }, i) => {
+          const style = {};
+          if (color) style['color'] = color;
```
Comment (dqunbp) [inline src/features/layer_features_panel/components/CardElements/Progress.tsx:31]: And here

File: src/features/layer_features_panel/components/CardElements/ActionButtons.tsx [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+
+export function ActionButtons({ items }: { items: ActionItem[] }) {
+  return (
+    <div className={s.actions}>
+      {items.map(({ title, type, data }, i): JSX.Element => {
+        let el;
+
+        if (type == 'external_link') {
+          el = (
+            <a key={i} href={data} target="_blank" rel="noreferrer">
```
Comment (dqunbp) [inline src/features/layer_features_panel/components/CardElements/ActionButtons.tsx:18]: noopener?

File: src/features/layer_features_panel/components/CardElements/Progress.tsx (hunk 10 above)
Comment (vkozio) [reply src/features/layer_features_panel/components/CardElements/Progress.tsx:22]: already used in data transformer

File: src/features/layer_features_panel/components/CardElements/Progress.tsx (hunk 11 above)
Comment (vkozio) [reply src/features/layer_features_panel/components/CardElements/Progress.tsx:31]: same

File: src/features/layer_features_panel/components/CardElements/ActionButtons.tsx (hunk 12 above)
Comment (vkozio) [reply src/features/layer_features_panel/components/CardElements/ActionButtons.tsx:18]: noreferrer already does that

File: src/features/layer_features_panel/atoms/hotProjects_outlines.ts
Comment (vkozio) [reply src/features/layer_features_panel/atoms/hotProjects_outlines.ts:1]: it's exactly layer id

**Review by Akiyamka — APPROVED** [review]

**Pull request merged**